python/graphstorm/dataloading/dataset.py

```diff
@@ -787,3 +790,74 @@ def infer_idxs(self):
         """ Set of nodes to do inference.
         """
         return self._infer_idxs
+
+class GSDistillData(Dataset):
+    """ Dataset for distillation
+
+    Parameters
+    ----------
+    file_list : list of str
+        List of input files.
+    tokenizer : transformers.AutoTokenizer
+        HuggingFace Tokenizer.
+    device : str
+        Device name.
+    """
+    def __init__(self, file_list, tokenizer, device):
+        super().__init__()
+        self.file_list = file_list
+        self.tokenizer = tokenizer
+        self.device = device
+        self.token_id_inputs, self.labels = self.get_inputs()
+
+    def get_inputs(self):
+        """ Tokenize textual data."""
+        for i, file_name in enumerate(self.file_list):
+            if i == 0:
+                inputs = pd.read_parquet(file_name)
+            else:
+                inputs = pd.concat([inputs, pd.read_parquet(file_name)])
```

Comment:
are there too many concatenations?

Reply:
No. This is to concat the whole dataframe from a file. Current implementation there can only be one file in the file_list.

Reply:
Who can control the number of files?

Reply:
file_list is defined from its caller. Currently I only put 1 file for each call.